### PR TITLE
Add version_id option to put_object_tagging/4

### PR DESCRIPTION
**Pull request by ishikawa**
Add `:version_id` option to `put_object_tagging/4` function. The API documentation is [here](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObjectTagging.html#API_PutObjectTagging_RequestSyntax).

This PR includes diffs by `mix format`. The main commit is f6d5d24ba3d9debaf4f2f40f003a911cc2e14787.

### CODE REVIEW

File: lib/ex_aws/s3.ex [hunk 1]
```diff
@@ -72,100 +72,103 @@ defmodule ExAws.S3 do
...
-  @type canned_acl :: :private
-    | :public_read
-    | :public_read_write
-    | :authenticated_read
-    | :bucket_owner_read
-    | :bucket_owner_full_control
-  @type grantee :: [ {:email, binary}
-    | {:id, binary}
-    | {:uri, binary}
-  ]
```
Comment (ishikawa) [inline lib/ex_aws/s3.ex:88]: I also apply `mix format` in afdf426. I'll comment about intentional changes.

File: lib/ex_aws/s3.ex [hunk 2]
```diff
@@ -753,45 +814,65 @@ defmodule ExAws.S3 do
...
   @spec put_object_tagging(
           bucket :: binary,
           object :: binary,
           tags :: Access.t(),
-          opts :: Keyword.t()
+          opts :: [option]
         ) :: ExAws.Operation.S3.t()
+        when option: {:version_id, binary} | {atom, binary}
   def put_object_tagging(bucket, object, tags, opts \\ []) do
+    {version_id, opts} = Keyword.pop(opts, :version_id)
```
Comment (ishikawa) [inline lib/ex_aws/s3.ex:867]: Take away `version_id` from options. It's okay because only `x-amz-expected-bucket-owner` and `content-md5` are only valid request headers.

File: test/lib/s3_test.exs [hunk 3]
```diff
@@ -435,6 +442,15 @@ defmodule ExAws.S3Test do
     assert expected == S3.put_object_tagging(bucket, object, test: "hello")
   end
 
+  test "#put_object_tagging w/ version_id" do
+    bucket = "my-bucket"
+    object = "test.txt"
+    version_id = "GOh7ob90QUq53H4Vd4aacioB6Nt.NoaU"
+
+    assert %Operation.S3{params: %{"versionId" => ^version_id}} =
+             S3.put_object_tagging(bucket, object, [test: "hello"], version_id: version_id)
```
Comment (ishikawa) [inline test/lib/s3_test.exs:451]: A new test for version_id.

File: lib/ex_aws/s3.ex [hunk 4]
```diff
@@ -848,14 +848,31 @@ defmodule ExAws.S3 do
...
+
+  """
   @spec put_object_tagging(
           bucket :: binary,
           object :: binary,
           tags :: Access.t(),
-          opts :: Keyword.t()
+          opts :: [option]
         ) :: ExAws.Operation.S3.t()
+        when option: {:version_id, binary} | {atom, binary}
```
Comment (bernardd) [inline lib/ex_aws/s3.ex:865]: I'm not quite clear why this change has been made. The type `{atom, binary}` should already cover `{:version_id, binary}`, and indeed both types are already covered by `Keyword.t()` as was in the original version.

**Comment by bernardd** [comment]
Hi @ishikawa - thanks for the PR! Overall it looks great, I've just got one question which I've added.

File: lib/ex_aws/s3.ex (hunk 4 above)
Comment (ishikawa) [reply lib/ex_aws/s3.ex]: @bernardd You're right 👍  I had missed it. Fixed in 478aa61.

Comment (bernardd) [reply lib/ex_aws/s3.ex]: Thanks - sorry to be a pain, but can you wind the typespec back to how it was? That is, `opts` just defined as a `Keyword.t()` rather than its own type? Then we're good to merge - thanks :)

Comment (ishikawa) [reply lib/ex_aws/s3.ex]: @bernardd Okay, reverted in 575465f
